## Setup

In [ ]:
import pandas as pd
pd.options.display.float_format = '{:.1f}'.format

In [ ]:
ufos_path = 'YOUR PATH HERE'
sightings = pd.read_csv(ufos_path)
sightings.head()

<ipython-input-3-828416dd11ab>:2: DtypeWarning: Columns (5,9) have mixed types. Specify dtype option on import or set low_memory=False.
  sightings = pd.read_csv(ufos_path)


datetime                  city state country     shape  \
0  10/10/1949 20:30            san marcos    tx      us  cylinder   
1  10/10/1949 21:00          lackland afb    tx     NaN     light   
2  10/10/1955 17:00  chester (uk/england)   NaN      gb    circle   
3  10/10/1956 21:00                  edna    tx      us    circle   
4  10/10/1960 20:00               kaneohe    hi      us     light   

  duration (seconds) duration (hours/min)  \
0               2700           45 minutes   
1               7200              1-2 hrs   
2                 20           20 seconds   
3                 20             1/2 hour   
4                900           15 minutes   

                                            comments date posted    latitude  \
0  This event took place in early fall around 194...   4/27/2004  29.8830556   
1  1949 Lackland AFB&#44 TX.  Lights racing acros...  12/16/2005    29.38421   
2  Green/Orange circular disc over Chester&#44 En...   1/21/2008        53.2   
3  My older brother and twin sister were leaving ...   1/17/2004  28.9783333   
4  AS a Marine 1st Lt. flying an FJ4B fighter/att...   1/22/2004  21.4180556   

   longitude  
0      -97.9  
1      -98.6  
2       -2.9  
3      -96.6  
4     -157.8

## What is the highest latitude of a sighting identified with a "shape" of cylinder?

In [ ]:
sightings['latitude'] = pd.to_numeric(sightings['latitude'], errors='coerce')

In [ ]:
cylinders = sightings[sightings['shape'] == 'cylinder']
cylinders['latitude'].max()

71.3875

**71.3875 degrees**

## How does the average duration of sightings in the United States compare to the median duration of sightings in Australia?

In [ ]:
sightings['duration (minutes)'] = pd.to_numeric(sightings['duration (seconds)'], errors='coerce') / 60

In [ ]:
us_sightings = sightings[sightings['country'] == 'us']
us_sightings['duration (minutes)'].mean()

96.66986747962486

In [ ]:
au_sightings = sightings[sightings['country'] == 'au']
au_sightings['duration (minutes)'].mean()

63.441153965303585

**Average duration of sightings in US: 96.67 minutes**

**Average duration of sightings in Australia: 63.44 minutes**

## Which city in the US state of California had the most sightings?

In [ ]:
ca_sightings = sightings[sightings['state'] == 'ca']
ca_sightings.describe(include='all')

datetime         city state country  shape duration (seconds)  \
count             9655         9655  9655    8912   9405               9655   
unique            9280         1806     1       1     23                259   
top     9/1/1999 21:30  los angeles    ca      us  light                300   
freq                12          352  9655    8912   1983                881   
mean               NaN          NaN   NaN     NaN    NaN                NaN   
std                NaN          NaN   NaN     NaN    NaN                NaN   
min                NaN          NaN   NaN     NaN    NaN                NaN   
25%                NaN          NaN   NaN     NaN    NaN                NaN   
50%                NaN          NaN   NaN     NaN    NaN                NaN   
75%                NaN          NaN   NaN     NaN    NaN                NaN   
max                NaN          NaN   NaN     NaN    NaN                NaN   

       duration (hours/min)               comments date posted  latitude  \
count                  9655                   9654        9655    9655.0   
unique                 1866                   9642         306       NaN   
top               5 minutes  Orange ball of light.  10/30/2006       NaN   
freq                    573                      3         186       NaN   
mean                    NaN                    NaN         NaN      35.8   
std                     NaN                    NaN         NaN       2.5   
min                     NaN                    NaN         NaN     -28.5   
25%                     NaN                    NaN         NaN      33.9   
50%                     NaN                    NaN         NaN      34.5   
75%                     NaN                    NaN         NaN      37.8   
max                     NaN                    NaN         NaN      61.9   

        longitude  duration (minutes)  
count      9655.0              9654.0  
unique        NaN                 NaN  
top           NaN                 NaN  
freq          NaN                 NaN  
mean       -119.4                65.5  
std           3.1              2261.6  
min        -124.5                 0.0  
25%        -121.7                 0.5  
50%        -118.8                 3.0  
75%        -117.9                10.0  
max          -6.2            175440.0

**Los Angeles**